crates/rpc-server/src/registry.rs

```diff
@@ -607,16 +635,44 @@ impl RequestSubmitter {
                     Ok(None) => Ok(()),
                     Err(err) => Err(err),
                 } {
+                    if let Some(TransactionError::BlockCyclesLimitReached { .. }) =
+                        err.downcast_ref::<TransactionError>()
+                    {
+                        log::info!("[tx from zero] mem block cycles limit reached, retry later");
+
+                        for (entry, handle) in items {
+                            self.queue.add(entry, handle);
+                        }
+                        continue;
+                    }
+
                     log::info!("[tx from zero] create account {}", err);
                 }
 
                 let snap = self.mem_pool_state.load();
                 let state = snap.state().expect("get mem state");
+                let mut block_cycles_limit_reached = false;
 
                 // Note: don't change `_handle` to `_`. The request should be
                 // removed from the in queue request map after pushed to mem
                 // pool.
-                for (entry, _handle) in items {
+                for (entry, handle) in items {
```

Comment:
The comment above is outdated now. Let's remove it.

Reply:
Ok